### PR TITLE
Fixing issue #1127 

### PR DESCRIPTION
**Pull request by louisdecharson**
Proposition to fix issue #1127

### CODE REVIEW

File: apps/oozie/src/oozie/views/dashboard.py [hunk 1]
```diff
@@ -968,7 +968,13 @@ def submit_external_job(request, application_path):
     params_form = ParametersFormSet(request.POST)
 
     if params_form.is_valid():
-      mapping = dict([(param['name'], param['value']) for param in params_form.cleaned_data])
+      if application_name == 'coordinator.xml':
```
Comment (romainr) [inline apps/oozie/src/oozie/views/dashboard.py:971]: mapping = {}

File: apps/oozie/src/oozie/views/dashboard.py [hunk 2]
```diff
@@ -968,7 +968,13 @@ def submit_external_job(request, application_path):
...
+        external_properties = Submission(request.user, fs=request.fs, jt=request.jt).get_external_parameters(application_path)
+        non_parameters = ParameterForm.get_non_parameters(external_properties)
+        mapping = non_parameters
+        mapping.update(dict([(param['name'], param['value']) for param in params_form.cleaned_data]))
+      else :
+        mapping = dict([(param['name'], param['value']) for param in params_form.cleaned_data])
```
Comment (romainr) [inline apps/oozie/src/oozie/views/dashboard.py:977]: Then outside of the if we do a single?

        mapping.update(dict([(param['name'], param['value']) for param in params_form.cleaned_data]))

Comment (louisdecharson) [reply apps/oozie/src/oozie/views/dashboard.py]: Yes, done my last commit [7386ff7](https://github.com/cloudera/hue/pull/1128/commits/7386ff7bddc49b80e97ad51adf630644dec9451f)

File: apps/oozie/src/oozie/views/dashboard.py [hunk 3]
```diff
@@ -968,6 +968,12 @@ def submit_external_job(request, application_path):
     params_form = ParametersFormSet(request.POST)
 
     if params_form.is_valid():
+      if application_name == 'coordinator.xml':
+        external_properties = Submission(request.user, fs=request.fs, jt=request.jt).get_external_parameters(application_path)
+        non_parameters = ParameterForm.get_non_parameters(external_properties)
+        mapping = non_parameters
```
Comment (romainr) [inline apps/oozie/src/oozie/views/dashboard.py:974]: Could have been merged but this is a nit and can be done later

**Comment by louisdecharson** [comment]
On a related note, it does not seem to be possible to view a workflow / coordinator stored in HDFS in workflow / coordinator editor (at least in CDH 6.2.1), is it ? Similarly, queries stored in HDFS view in Query editor can't be saved in HDFS when modified in Query editor.

Is there any work-stream on the topic or theoretical reasons for not going in that direction ?

I see many good reasons for working with HDFS files rather than documents : they can be shared more easily and it also make one's life easier when working with queries / workflows / coordinators that need to be duplicated across multiple environnements (Acceptance, Production, etc).

**Comment by romainr** [comment]
"Similarly, queries stored in HDFS view in Query editor can't be saved in HDFS when modified in Query editor."

Indeed, the feature was never fully implemented (only read from the File System, HDFS, S3... was done) but allowing to save back would be a nice contribution as it is handy and simple to open queries directly there. It was just not implemented so far (same as doing the same via Git).

**Comment by louisdecharson** [comment]
Hi @romainr,

I was wondering : is there any way to import an oozie workflow (with `workflow.xml`, `job.properties`, `lib/` folder, etc) to the Oozie editor ?

Cheers,

Louis

**Comment by romainr** [comment]
It used to be possible until version 3.8, but as parsing the workflows (with XST etc) was 95% correct but many little nits were missing and we were getting too many complains from paying customers so the option got hidden. It was only .xml, but would not be too hard to also add properties and files from a zip.

The code might still around https://github.com/cloudera/hue/tree/master/apps/oozie/src/oozie/importlib

More background:
https://www.google.com/search?q=hue+oozie+xml+import&oq=hue+oozie+xml+import&aqs=chrome..69i57j0.3224j0j1&sourceid=chrome&ie=UTF-8

**Pull request merged**